File: 645.SetMismatch/mismatch.py
class Solution(object):
    def findErrorNums(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        counts = [0] * len(nums)
        ans = []
        for n in nums:
            counts[n-1] += 1
            if counts[n-1] > 1:
                ans.append(n)

        for i in range(len(counts)):
            if counts[i] == 0:
                ans.append(i+1)

        return ans

if __name__ == '__main__':
    nums = [1,2,2,4]
    print Solution().findErrorNums(nums)

File: 645.SetMismatch/Mismatch.java
class Solution {
    public int[] findErrorNums(int[] nums) {
        int[] counts = new int[nums.length];
        int[] ans = new int[2];
        for (int n : nums) {
            counts[n-1]++;
            if (counts[n-1] > 1) {
                ans[0] = n;
            }
        }
        for (int i = 0; i < counts.length; i++) {
            if (counts[i] == 0) {
                ans[1] = i + 1;
                break;
            }
        }
        return ans;
    }
}
